experimental/devices/bitbang/spi.go

```diff
@@ -147,36 +150,97 @@ func (s *spiConn) Duplex() conn.Duplex {
 	return conn.Full
 }
 
+func (s *spiConn) clockOn() error {
+	if s.mode&spi.Mode2 == spi.Mode2 {
```

Comment:
One option is to add properties to spiConn for clock, i.e. at line 135, replace mode by:

    clockIdle           gpio.Level
    readAfterClockPulse bool

and initialize them in Connect(). Then you can change the clockOn/clockOff functions to be:

    s.sck.Out(s.clockIdle)
    s.sck.Out(!s.clockIdle)

Reply:
That sounds nicer, yes.